components/pkg-aci/bin/hab-pkg-aci.sh

```diff
@@ -150,28 +154,28 @@ build_aci() {
 # The root of the filesystem. If the program is running on a separate
 # filesystem or chroot environment, this environment variable may need to be
 # set.
-: ${FS_ROOT:=}
+: "${FS_ROOT:=}"
 # The root path of the Habitat file system. If the `$HAB_ROOT_PATH` environment
 # variable is set, this value is overridden, otherwise it is set to its default
-: ${HAB_ROOT_PATH:=$FS_ROOT/hab}
+: "${HAB_ROOT_PATH:=$FS_ROOT/hab}"
 # Directory to write ACIs to. If $HAB_RESULTS_DIR environment variable
 # is set, this value is overridden
-: ${HAB_RESULTS_DIR:=/src/results}
+: "${HAB_RESULTS_DIR:=/src/results}"
 
 # The current version of Habitat Studio
 version='@version@'
 # The author of this program
 author='@author@'
 # The short version of the program name which is used in logging output
-program=$(basename $0)
+program=$(basename "$0")
 
 find_system_commands
 
-if [ -z "$@" ]; then
+if [ -z "$*" ]; then
```

Comment:
https://github.com/koalaman/shellcheck/wiki/SC2198

In general, the `[` conditions are all either unary or binary and passing the wrong number of tokens to them is an error. `-z` is unary, so instead of `"$@"`, which expands to multiple tokens, we must use `"$*"` to get a single string.